chajda/tsquery/augments.py

```diff
@@ -52,62 +52,185 @@ def augments_gensim(lang, word, config=Config(), n=5):
+        try:
+            annoy_indices[lang].load(f"{chajda_path}/annoy/{lang}.ann")
+
+        # OSError occurs if index is not already saved to disk
+        except OSError:
+            # The annoy library uses a random number genertor when building up the trees for an Annoy Index.
+            # In order to ensure that the output is deterministic, we specify the seed value for the random number generator.
+            annoy_indices[lang].set_seed(22)
+
+            # Populate AnnoyIndex with vectors from corresponding fasttext model
+            i = 0
+            for j in fasttext_models[lang].words:
+                v = fasttext_models[lang][j]
+                annoy_indices[lang].add_item(i,v)
+                i += 1
+
+            # build the trees for the index    
+            annoy_indices[lang].build(10)
+
+            # save the index to annoy directory,
+            # make the directory if needed
+            try:
+                annoy_indices[lang].save(f"{chajda_path}/annoy/{lang}.ann")
+            except OSError:
+                os.mkdir(f"{chajda_path}/annoy")
+                annoy_indices[lang].save(f"{chajda_path}/annoy/{lang}.ann")
+
+
+def load_fasttext_model(lang, word='google', n=5):
+    '''
+    Downloads and loads a fasttext model for a particular language,
+    storing the fasttext model object in the fasttext_models dictionary.
+    Due to the size of fasttext models (>1gb), and the loading time,
+    we do not load all models on program startup.
+    Instead, this function is used to load them lazily as needed. 
+
+    lang: language passed from augments_fasttext
+    
+    # >>> load_fasttext_model('en') or fasttext_models['en'] != None
+    # True
+    '''
+    # suppress_stdout_stderr() is used for redirecting stdout and stderr when downloading and loading fasttext models.
+    # Without this, stdout and stderr from fasttext causes doctests to fail. 
+    with suppress_stdout_stderr():
+
+        # fasttext downloads models to the current directory,
+        # but we want to store the model in chaja/tsquery/fasttext,
+        # so we change the working directory before downloading
+        try:      
+            os.chdir(f"{chajda_path}/fasttext")
+        except OSError:
+            os.mkdir(f"{chajda_path}/fasttext")
+            os.chdir(f"{chajda_path}/fasttext")
+
+        # download and load fasttext model
+        fasttext.util.download_model(lang, if_exists='ignore')
+        fasttext_models[lang] = fasttext.load_model(f"cc.{lang}.300.bin")
+
+        # change working directory back
+        os.chdir(working_dir)
+
+    # Executing fasttext's get_nearest_neighbor as well as accessing the words in the fasttext model both run
+    # significantly faster after they has been called once,
+    # so we call them once upon loading the model to ensure most efficient runtime for subsequent uses.
+    fasttext_models[lang].get_nearest_neighbors(word, k=n)
```

Comment:
`word` and `n` should not be parameters to the function, but hard-coded here; this is different than the `seed` because these are an internal implementation "wart" of your code that the user shouldn't be exposed to and providing different values will not change the behavior of the function in any way; `seed`, however, is a standard thing that needs to be set in many situations and is something that can change the behavior of the function, and so some users may need to specify the seed

Reply:
actually, the most correct thing to do with the seed would be to include the seed in the filename of the annoy index that you save to disk, and then load/save the file based on the seed; this would allow a user to create multiple indexes with multiple seeds and see how that affects the performance of the model; in practice, however, no one would want to do this, so don't feel a need to actually add this feature (unless you want to), but this is a standard thing I do when creating any files that depend on a seed just in case